assembly/src/parsers/ast/serde/serialization.rs

```diff
@@ -51,6 +51,19 @@ impl ByteWriter {
         Ok(())
     }
 
+    pub fn write_docs(&mut self, val: &Option<String>) {
+        match val {
+            Some(docs) => {
+                let doc_bytes = docs.as_bytes();
+                self.write_u16(doc_bytes.len() as u16);
```

Comment:
Might be good to check if the length of the string is actually smaller than `u16::MAX_VALUE`, and return an error if it is not.